lib/util/regex.ts

```diff
@@ -19,17 +19,25 @@ try {
   RegEx = RegExp;
 }
 
-export function regEx(pattern: string | RegExp, flags?: string): RegExp {
-  const key = `${pattern.toString()}:${flags}`;
+export function regEx(
+  pattern: string | RegExp,
+  flags?: string,
```

Comment:
```suggestion
  flags?: string | undefined,
```
Required for strict null checks